text/3064-bitfield.md

```diff
@@ -192,7 +192,7 @@ If the width of the bit-field is `0`, then the name of the field must be `_`.
 
 A field named `_` is called an *unnamed* field. All other fields are called
 *named* fields. Each named field annotated with `bitfield(N)` occupies `N` bits
-of storage.
+of storage. Unnamed fields do not occupy any storage.
```

Comment:
This doesn't really clarify anything, the term "storage" is really ambiguous and the wording of "unnamed fields do not occupy any storage" implies that unnamed fields don't take up any space whatsoever, which isn't correct

Reply:
>implies that unnamed fields don't take up any space whatsoever, which isn't correct

That is the intended reading. Unnamed fields only affect the layout but are not themselves members of the struct. See also the *equivalent* notation for unnamed fields using attributes.

In C++, unnamed fields are described as "not being members" of their surrounding struct. In C the language is not as simple but this difference is hopefully irrelevant in practice.

Reply:
The space still exists though, those bits don't just disappear right?

Reply:
I do not know what you mean by "the space exists". Space between fields in structs "exists" but its the behavior of that space is not defined in great detail as far as I can tell.

Reply:
Right, that's the problem. You have to say what's going on with that. Even if C doesn't, in Rust we care about those sorts of things.

Reply:
>You have to say what's going on with that. Even if C doesn't, in Rust we care about those sorts of things.

Please point me to the part of the current reference that documents the behavior of padding bits in repr(C) structs or any kind of struct.

Reply:
[there's actually quite a few issues about it](https://github.com/rust-lang/unsafe-code-guidelines/issues?q=is%3Aissue+is%3Aopen+padding)

Reply:
This RFC is not the place to resolve these issues.

Reply:
This issue seems to be the most relevant: https://github.com/rust-lang/unsafe-code-guidelines/issues/156

Reply:
You don't have to fix all of uninit memory for rust, you just have to say something about the spare bits in this one case.

It can be "the bits are unspecified" (my pick), or "the bits are niche", or "the bits are always zero", or whatever, but the RFC shouldn't leave this unanswered.

Reply:
If you get such language merged into the language reference for padding bits in general, then I will refer to it. Otherwise I do not see why this RFC should specify the state of padding bits of

```rust
#[repr(C)]
struct X {
	a: u8,
	#[bitfield(8)] _: u8,
	b: u16,
}
```

when the state of padding bits

```rust
#[repr(C)]
struct X {
	a: u8,
	b: u16,
}
```

is not specified.

Reply:
Yah, this RFC is in a position in which it can entirely bypass the issues of previous code with uninit memory since it can simply declare what the state of "padding bits" should be, which should almost definitely be unspecified